weave/weave_client.py

```diff
@@ -142,6 +142,8 @@ def map_to_refs(obj: Any) -> Any:
         return [map_to_refs(v) for v in obj]
     elif isinstance(obj, dict):
         return {k: map_to_refs(v) for k, v in obj.items()}
+    elif isinstance(obj, WeaveObject):
```

Comment:
might be nice to add a little comment and say that this is only expected under the condition that the WeaveObject was previously dirtied